pkg/common/ipsec/utils.go

```diff
@@ -4,14 +4,83 @@
 package ipsec
 
 import (
+	"encoding/hex"
 	"errors"
 	"fmt"
 
 	"github.com/vishvananda/netlink"
 
 	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/datapath/linux/safenetlink"
 )
 
+// XfrmStateInfo represents the key information from an XFRM state
+// This struct is used for JSON serialization of XFRM state information
+// and is shared between cilium-dbg and cilium-cli for consistency
+type XfrmStateInfo struct {
+	Src      string `json:"src"`
+	Dst      string `json:"dst"`
+	SPI      uint32 `json:"spi"`
+	ReqID    uint32 `json:"reqid"`
+	AuthAlg  string `json:"auth_alg,omitempty"`
+	AuthKey  string `json:"auth_key,omitempty"`
+	CryptAlg string `json:"crypt_alg,omitempty"`
+	CryptKey string `json:"crypt_key,omitempty"`
+	AeadAlg  string `json:"aead_alg,omitempty"`
+	AeadKey  string `json:"aead_key,omitempty"`
+}
+
+// DumpXfrmStates extracts XFRM state information using netlink
+func DumpXfrmStates() ([]XfrmStateInfo, error) {
```

Comment:
Why is this introduced in the first commit just to be removed in the second? Please spend more time checking your own changes.